### DISCUSSION TITLE
supabase-auth-token” is invalid because its size is too big. Max size is 4096 B.

### DISCUSSION THREAD

**Discussion by lvanoverberghe** [supabase/supabase#9972] [Questions]
I tried to implement the azure login and the user is created in the database but I am getting this error in the Firefox console.

**Comment by shanehandley** [comment]
Have just coma across this issue - in my case I was following the pattern of creating `auth.users` rows with `raw_user_metadata` whcih was used in a trigger to populate a `profiles` row, along the lines of this:

https://supabase.com/docs/guides/auth/managing-user-data#using-triggers

The issue with this is that when the user is authenticated, supabase returns the `user` row in the response, which includes all `raw_user_metadata` used during registration. Supabase tries to write this information to a cookie but given that `raw_user_metadata` can include a substatial amount of information, it can cause the cookie size to exceed 4kb.

The solution in my case was to change the trigger function to drop that `raw_user_metadata` after it had been used to create the `profile`. For example, based on the hook code in the above link:

```sql
-- inserts a row into public.profiles
create function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.profiles (id, first_name, age)
  values (new.id, new.raw_user_meta_data ->> 'first_name', new.raw_user_meta_data['age']::integer);
  
  -- Added --> once this data has been inserted into the profiles table, we no longer need to retain it in the auth.users table
  update auth.users set raw_user_meta_data = null where id = new.id;
  
  return new;
end;
$$;

-- trigger the function every time a user is created
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```
